packages/stockflux-launcher/src/App.js

```diff
@@ -1,66 +1,146 @@
-import React, { useRef, useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import cx from 'classnames';
 import { OpenfinApiHelpers } from 'stockflux-core';
-import { useDockWindow, ScreenEdge } from 'openfin-react-hooks';
+import {
+  useDocked,
+  useDockWindow,
+  useOptions,
+  ScreenEdge
+} from 'openfin-react-hooks';
 import {
   FaChevronUp,
   FaChevronLeft,
   FaChevronRight,
-  FaRegHandRock
+  FaRegHandRock,
+  FaUnlock
 } from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
 import ToolBar from './toolbar/ToolBar';
 import CloseButton from './toolbar/CloseButton';
 import './App.css';
 
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
-
 export default () => {
+  const [isDockable, setIsDockable] = useState(false);
+  const [isDocked, setDocked] = useState(true);
+  const [isHorizontal, setHorizontal] = useState(true);
+  const [options] = useOptions();
+  const [undockTop, setUndockTop] = useState(0);
+  const [undockLeft, setUndockLeft] = useState(0);
```

Comment:
I'm not sure what we are achieving with these variables. Why we are only storing `undockLeft`, but not `undiockRight`? Could this not be covered by looking up which edge the launcher is docked to (line 30)? If this is truly needed then I would suggest having a single state variable:
```
const [undock, setUndock] = useState(
  {
    left: 0,
    top: 0
  }
)
```

Reply:
`undockLeft` is stored but not `undockRight` as `undockLeft` is used to set the position of the window when undocked. But agree that the state can be simplified into an `undockPosition` object, which should hopefully make its use more obvious at a glance.